x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/FrequentItemSetsAggregationBuilder.java

```diff
@@ -0,0 +1,168 @@
+        }
+    );
+
+    static {
+        ContextParser<Void, MultiValuesSourceFieldConfig.Builder> metricParser = MultiValuesSourceFieldConfig.parserBuilder(
+            false, // scriptable
+            true,  // timezone aware
+            false, // filtered
+            true   // format
+        );
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(), (p, n) -> metricParser.parse(p, null).build(), FIELDS);
+        PARSER.declareDouble(ConstructingObjectParser.optionalConstructorArg(), MINIMUM_SUPPORT);
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), MINIMUM_SET_SIZE);
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), Aggregation.CommonFields.SIZE);
+    }
+
+    private final List<MultiValuesSourceFieldConfig> fields;
+    private final double minimumSupport;
+    private final int minimumSetSize;
+    private final int size;
+
+    public FrequentItemSetsAggregationBuilder(
+        String name,
+        List<MultiValuesSourceFieldConfig> fields,
+        double minimumSupport,
+        int minimumSetSize,
+        int size
+    ) {
+        super(name);
+        this.fields = fields;
+        if (minimumSupport <= 0 || minimumSupport > 1) {
+            throw new IllegalArgumentException(
+                "[minimum_support] must be greater than 0 and less or equal to 1. Found [" + minimumSupport + "] in [" + name + "]"
+            );
+        }
+        this.minimumSupport = minimumSupport;
+        if (minimumSetSize <= 0) {
+            throw new IllegalArgumentException(
+                "[minimum_set_size] must be greater than 0. Found [" + minimumSetSize + "] in [" + name + "]"
+            );
+        }
+        this.minimumSetSize = minimumSetSize;
+        if (size <= 0) {
+            throw new IllegalArgumentException("[size] must be greater than 0. Found [" + size + "] in [" + name + "]");
+        }
+        this.size = size;
+    }
+
+    public FrequentItemSetsAggregationBuilder(StreamInput in) throws IOException {
+        super(in);
+        this.fields = in.readList(MultiValuesSourceFieldConfig::new);
+        this.minimumSupport = in.readDouble();
+        this.minimumSetSize = in.readInt();
+        this.size = in.readInt();
+    }
+
+    @Override
+    protected AggregationBuilder shallowCopy(Builder factoriesBuilder, Map<String, Object> metadata) {
+        return new FrequentItemSetsAggregationBuilder(name, fields, minimumSupport, minimumSetSize, size);
+    }
+
+    @Override
+    public String getType() {
+        return NAME;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeList(fields);
+        out.writeDouble(minimumSupport);
+        out.writeInt(minimumSetSize);
+        out.writeInt(size);
+    }
+
+    @Override
+    protected AggregatorFactory doBuild(AggregationContext context, AggregatorFactory parent, Builder subfactoriesBuilder)
```

Comment:
I think we should indicate that it is indeed a `MANY` bucket cardinality and also ensure `subfactoriesBuilder` is empty.